docs/design/submarine-server/architecture.md

```diff
@@ -0,0 +1,126 @@
+                 |    | |      | | | YARN | |  K8s  | | |   |   | +--------+   +----+ |
+ +-----------+   |    | |      | | +------+ +-------+ | |   |   |     YARN Cluster    |
+ |           |   |    | |      | |     submitter      | |   |   +---------------------+
+ |    CLI    +------->+ |      | +--------------------+ +---+
+ |           |   |    | | REST | +--------+ +---------+ |   |   +---------------------+
+ +-----------+   |    | |      | | TRANSL | | monitor | |   |   | +--------+   +----+ |
+                 |    | |      | +--------+ +---------+ |   |   | |        +-->+job1| |
+ +-----------+   |    | |      | +--------------------+ |   |   | |        |   +----+ |
+ |           |   |    | |      | |     JobManager     | |   +-->+ |operator|   +----+ |
+ |    SDK    +---+    | +------+ +--------------------+ |       | |        +-->+job2| |
+ |           |        +---------------------------------+       | +--------+   +----+ |
+ +-----------+                                                  |     K8s Cluster     |
+    client                          server                      +---------------------+
+ ```
+We propose to split the core module to two modules, CLI and server as shown in FIG. The client call the REST API to submit and retrieve the job info. 
+
+### Submarine module structure
+```
+         |--submarine
+	     |--submarine-all
+   	     |--submarine-client
+   	     |--submarine-commons
+	     |--submarine-dist
+	     |--submarine-sdk
+	     |    |--pysubmarine
+	     |--submarine-server
+	     |    |--server-core
+	     |    |--server-submitter
+	     |    |   |----submitter-yarn
+	     |    |   |----submitter-yarnservice
+	     |    |   |----submitter-k8s
+	     |    |--server-yarn-runtime
+	     |    |--server-operator
+	     |--submarine-workbench
+	     |    |--workbench-server
+	     |    |--workbench-web
+	     |--submodules
+```
+
+### submarine-client
+The client as the default implements for submarine server RESTful API, it provides all the features which declared in the RESTful API.
+
+### submarine-server
+```
++------+ +--------------------+
+|      | | +------+ +-------+ |
+|      | | | YARN | |  K8s  | |
+|      | | +------+ +-------+ |
+|      | |     submitter      |
+|      | +--------------------+
+| REST | +--------+ +---------+
+|      | | TRANSL | | monitor |
+|      | +--------+ +---------+
+|      | +--------------------+
+|      | |     JobManager     |
++------+ +--------------------+
+```
+The project structure is as follows:
+```
+	|--submarine-server
+	|----server-api
+	|----server-core
+	|----server-submitter
+	|----pom.xml
+	|----README
+```
+The server-sdk will be added in the future. It is the official client for Submarine Server API.
+
+#### REST
+It provides the RESTful API for users, consists of submit job and get job info features and so on. Often used by CLI and workbench etc. The submarine client project should implement all APIs.
```

Comment:
Can we define the REST APIs here? When defining it, we should consider the current YAML files format and the tf-operator job spec, pyTorch operator job spec and also horovod spec. Please draft it if you've extra cycles and let's review it together.

Reply:
As the Birds Eyes View Architecture, I think we should keep it simple and should clever describe the architecture.  About the each component's detailed design maybe we can propose inside alone doc in the same location.